src/components/AddItems.js

```diff
@@ -36,6 +40,24 @@ const AddItemsToList = () => {
       .get()
       .then((data) => {
         if (data.docs.length) {
+          const { items } = data.docs[0].data();
+          const nonWordCharactersAndUnderscores = /[\W_]/g;
+          const shoppingListItemExists = items.some(
+            (shoppingListItemObject) => {
+              return (
+                shoppingListItemObject.shoppingListItemName
+                  .toLowerCase()
+                  .replace(nonWordCharactersAndUnderscores, '') ===
+                shoppingListItemName
+                  .toLowerCase()
+                  .replace(nonWordCharactersAndUnderscores, '')
+              );
+            },
+          );
+          if (shoppingListItemExists === true) {
+            setShoppingListItemNameExists(true);
```

Comment:
Until at this point where the state is changed to true. Aside from this observation every other thing is quite clear and nicely done. well done ones more

Reply:
Thanks